public/app/features/dashboard-scene/serialization/ElementPanelMappingService.ts

```diff
@@ -0,0 +1,60 @@
+// function to manage a lookup table that will hold element_identifer : panel_id
+
+// function to manage a lookup table that will hold element_identifer : panel_id
+export interface ElementPanelMapping {
+  [elementIdentifier: string]: number;
+}
+
+export class ElementPanelMappingService {
+  private static instance: ElementPanelMappingService;
+  private mapping: ElementPanelMapping = {};
```

Comment:
The mapping itself conceptually is domain of serialization/deserialisation of dashboard, at least in v2. I think it should belong to an abstraction that's responsible for it, which is `DashboardSceneSerializer`. With that we could also implement the siimilar logic for v1  through the serializers and only rely on the mapping there for the viz panel elements keys, rather then creating the keys dynamically when viz panels are rendered. Can we create the panel keys (uid) upfront when instantiating the scene and refer to them later on when rendering the VizPanels?

Reply:
My point is really, for v1 we create "fake uids", while for v2 we derive the uids from the `elements` keys. I think this is the first step for relying on the string identifier in the runtime instead of panel ID.